tests/integration-tests/tests/byos/test_byos.py

```diff
@@ -9,19 +9,172 @@
 # or in the "LICENSE.txt" file accompanying this file.
 # This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
 # See the License for the specific language governing permissions and limitations under the License.
+import json
 import logging
+import time
+from copy import deepcopy
 
+import boto3
 import pytest
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from retrying import retry
+from time_utils import minutes, seconds
 
 from tests.common.assertions import assert_head_node_is_running
 
+LAUNCH_TEMPLATES_CONFIG_PATH = "/opt/parallelcluster/shared/launch_templates_config.json"
+BYOS_SUBSTACK_OUTPUTS_PATH = "/opt/parallelcluster/shared/byos_substack_outputs.json"
+BYOS_LOG_PATH = "/var/log/parallelcluster/byos-plugin.log"
+BYOS_HOME = "/home/byos"
+BYOS_USERS_LIST = ["user1", "byosUser"]
+
 
 @pytest.mark.usefixtures("instance", "scheduler", "os")
-def test_byos(region, pcluster_config_reader, clusters_factory):
+def test_byos(region, pcluster_config_reader, s3_bucket_factory, clusters_factory, test_datadir):
     """Test usage of a custom scheduler integration."""
     logging.info("Testing custom scheduler integration.")
 
-    cluster_config = pcluster_config_reader()
+    # Create bucket and upload resources
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    for file in ["byos_infra.cfn.yaml", "artifact"]:
+        bucket.upload_file(str(test_datadir / file), f"byos/{file}")
+    # Create cluster
+    byos_user_list = deepcopy(BYOS_USERS_LIST)
+    cluster_config = pcluster_config_reader(bucket=bucket_name, user1=byos_user_list.pop(), user2=byos_user_list.pop())
```

Comment:
nit: if you want to get rid of the copy you can just assign byos_user_list[0] and byos_user_list[1]

Reply:
done